sources/vault.move

```diff
@@ -0,0 +1,122 @@
+/// User can deposit/withdraw multiple fungible assets to/from different markets
+module playground::vault {
+    use std::signer;
+    use std::string;
+
+    use aptos_std::table::{Self, Table};
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, Metadata};
+    use aptos_framework::object::{Self, Object};
+
+    #[test_only]
+    use aptos_framework::primary_fungible_store;
+    #[test_only]
+    use std::option;
+    use std::vector;
+
+    const ERR_UNAUTHORIZED: u64 = 1;
+    const ERR_INSUFFICIENT_FUNDS: u64 = 2;
+
+    /// Market holds user deposited assets
+    struct Market has key {
+        assets: Object<FungibleAsset>
+    }
+
+    /// Balance keeps track of user assets in different markets
+    struct User has key {
+        markets: Table<Object<Market>, u64>
+    }
+
+    public fun init_market(admin: &signer, asset: Object<Metadata>): Object<Market> {
+        assert!(signer::address_of(admin) == @playground, ERR_UNAUTHORIZED);
+
+        let asset_name = fungible_asset::name(asset);
+
+        let fa_seed = vector::empty<u8>();
```

Comment:
Another patter for doing this that looks better to me is to have a single object (instead of 2) that has both the FungibleAsset store (via create_object) + add the market resource there at the same time (with the resource group annotation)

Reply:
sry i am not following, since the `Market` resource is currently only holding the `FungibleAsset` object.

single object has both the FungibleAsset store + add the market resource there - what does it really mean?

Reply:
Each object is similar to an account and can hold multiple resources. Therefore, you can create a single Market object that has the following two things:
1. A fungible asset balance (FungibleAsset)
2. A Market resource that contains data such as total debt.
In the controller module where you need to evaluate cross-market liquidity, you can refer to each market as Object<Market>. You can even have a vector of markets - vector<Object<Market>> (or SmartVector).